mlir/utils/jenkins/Jenkinsfile.migraphxintegration

```diff
@@ -2,161 +2,209 @@ boolean shouldRunOnMachine(String machine){
     switch (machine){
         case "rocm-framework-38.amd.com":
             return (params.MI200 == true)
-        case "aus-navi3x-13.amd.com":
+        case "sh5-1e707-rf06-26.mkm.dcgpu":
+            return (params.MI300 == true)
+        case "supermicro-829":
+            return (params.Navi2x == true)
+        case "aus-navi3x-19.amd.com":
             return (params.Navi3x == true)
+        case "navi4x-hw-31.amd.com":
+            return (params.Navi4x == true)
     }
 }
 
 void clean() {
-sh """ #!/bin/bash -x
-
-if [ \$(docker ps -a -q -f name=migraphx) ]; then
-    docker stop migraphx
-    docker rm migraphx
-    echo "cleanup"
-fi
-"""
+    sh """ #!/bin/bash -x
+
+    if [ \$(docker ps -a -q -f name=migraphx) ]; then
+        docker stop migraphx
+        docker rm migraphx
+        echo "cleanup"
+    fi
+    """
+}
+
+void mountModels() { 
+    sh '''#!/bin/bash -x
+    set -e
+    set -o pipefail
+
+    if ! command -v sshfs &> /dev/null
+    then
+        echo "sshfs not installed, installing now."
+        sudo apt-get install -y sshfs
+    else
+        echo "sshfs is already installed."
+    fi
+
+    if ! command -v sshpass &> /dev/null
+    then
+        echo "sshpass not installed, installing now."
+        sudo apt-get install -y sshpass
+    else
+        echo "sshpass is already installed."
+    fi
+
+    echo "Setting variables for SSHFS..."
+    sshfs_user="$SSHFS_USER"
+    sshfs_pass="$SSHFS_PASS"
+    sshfs_host="mlse-nas.amd.com"
+    sshfs_remote_path="migraphx/models"
+    sshfs_local_path="/mnt/nas_share/migraphx/models"
+    sshfs_port="22"
+
+    known_hosts_file="/tmp/known_hosts"
+    ssh-keyscan -p "$sshfs_port" "$sshfs_host" > "$known_hosts_file"
```

Comment:
... I'd argue having the public key in by Jenkins credential is a better idea than trusting a keyscan

Reply:
The Tuna folks like to use Jenkins environment variables for these things, so they can set them up in the job and keep them out of github.  IT may also complain, even just about visible hostnames.

Our codecov stage uses `withCredentials` to fetch a Jenkins credential into an environment varlable.

My unpublished efforts to use an ssh tunnel for a Tuna database connection settled on `sshagent (credentials: ['rocmlir-ixt-rack-15']) { ... ssh commands ... }`

But note that ssh is weird on our CI nodes -- lockharts are different from 10.216.64.100 hosts are different from the rest.